### ISSUE TITLE
PolynomialFeatures can never be chosen

### ISSUE THREAD

**Issue by reeseb777** [rhiever/tpot#454]
The condition in the code that is intended to limit the number of PolynomialFeatures stages inadvertently prevents all pipelines with PolynomialFeatures from being considered.

## Context of the issue

the cause of this bug is base.py line 708:
             if individual_str.count('PolynomialFeatures') > 1:

the problem is that it disqualifies valid pipelines, such as:
LogisticRegression(PolynomialFeatures(input_matrix, PolynomialFeatures__degree=2, PolynomialFeatures__include_bias=DEFAULT, PolynomialFeatures__interaction_only=False), LogisticRegression__C=10.0, LogisticRegression__dual=DEFAULT, LogisticRegression__penalty=DEFAULT)

because even though there is only one PolynomialFeatures stage, the string 'PolynomialFeatures' appears multiple times.

[the remaining entries are only necessary if you are reporting a bug]

## Process to reproduce the issue

classifier_config_dict = {
    'sklearn.linear_model.LogisticRegression': {
        'penalty': ["l2"],
        'C': [10.],
        'dual': [True]
    },
    'sklearn.preprocessing.PolynomialFeatures': {
        'degree': [2],
        'include_bias': [False],
        'interaction_only': [False]
    },
    'sklearn.feature_selection.SelectKBest': {
        'k': range(1, 100), # need check range!
        'score_func': {
            'sklearn.feature_selection.f_classif': None
            }
    }
}

1. User creates TPOT instance using the above config dictionary
2. User calls TPOT `fit()` function with training data (verbosity=3)
3. TPOT reports many 'Invalid pipeline encountered. Skipping its evaluation.' messages

## Expected result

A pipeline with a single PolynomialFeatures stage is not invalid.

## Current result

A pipeline with a single PolynomialFeatures stage is invalid.

## Possible fix

change base.py line 708 to:
             if individual_str.count('PolynomialFeatures(') > 1:

optionally, add more verbose output that lets the user know why the pipeline is invalid

**Comment by weixuanfu2016** [comment]
Oh, right, this is a bug. I will submit a PR soon to fix it.

**Comment by weixuanfu2016** [comment]
I just posted a PR for fixing this issue. It is not merged to master branch now. If you need fix the tpot in your local environment for running some urgent jobs. Please try the command below to reinstall tpot from the fixed branch.

```
pip install --upgrade --no-deps --force-reinstall git+https://github.com/weixuanfu2016/tpot.git@issue454```

**Comment by rhiever** [comment]
Fixed and rolled out in TPOT 0.7.5. Thank you for the bug report.

**Comment by reeseb777** [comment]
Thanks a bunch for the speedy fix!
